umich_sim/sim_backend/config.py

```diff
@@ -0,0 +1,119 @@
+"""
+Implementation of the configuration class according to "Configuration Class Proposal"
+https://docs.google.com/document/d/12WPOGTSmqdKhLk_f6ysbAbLGbMhUN15yMLg7rpNrrJY/
+"""
+
+from typing import *
+from enum import Enum
+from dataclasses import dataclass
+
+# Enums (Auto-assign integer values starting with 1)
+# Environment
+Task = Enum('Task', ['INTERSECTION', 'EXPRESSWAY'])
+Weather = Enum('Weather', ['SUNNY', 'CLOUDY', 'RAINY', 'SNOWY'])
+Scene = Enum('Scene', ['DAY', 'NIGHT'])
+TrafficLightColor = Enum('TrafficLightColor', ['GREEN', 'YELLOW', 'RED'])
+TrafficLightTrigger = Enum('TrafficLightTrigger', ['TIME', 'DISTANCE'])
+Direction = Enum('Direction', ['STRAIGHT', 'LEFT', 'RIGHT'])
+Accident = Enum('Accident', ['NORMAL_STOP, ABRUPT_STOP', 'SLOW_INTRUDE', 'RUN_RED_LIGHT'])
+FreewayVehicleType = Enum('FreewayVehicleType', ['LEAD', 'AHEAD_LEAD', 'ADJACENT_LEAD', 'FOLLOW', 'SPEED_BARRIER'])
+FreewayVehicleBehavior = Enum('FreewayVehicleBehavior', ['KEEP_SAME_SPEED', 'AC_OR_DECELERATE', 'CHANGE_LANES', 'PULL_OFF_STOP', 'ENTER_ADJACENT'])
+# Vehicle
+VehicleType = Enum('VehicleType', [])  # to be specified
+VehicleColor = Enum('VehicleColor', [])  # to be specified
+TrafficRule = Enum('TrafficRule', ['FOLLOW', 'INTRUDE', 'IGNORE'])
+Route = Enum('Route', ['CHANGE_LANE', 'TURN_AT_INTERSECTION'])  # more to be specified
+
+@dataclass
+class Location:
+    # To be specified
+    pass
+
+# Environment -> intersections (list of Intersection) -> Intersection -> traffic_light
+@dataclass
+class TrafficLight:
+    state_durations: Dict[TrafficLightColor, float]
+    trigger_type: TrafficLightTrigger
+    trigger_val: float
+
+# Environment -> intersections (list of Intersection) -> Intersection -> vehicles (list of IntersectionVehicle)
+@dataclass
+class IntersectionVehicle:
+    init_location: Location
+    model: str
+    direction: Direction
+    accident: Accident
+
+# Environment -> intersections (list of Intersection)
+@dataclass
+class Intersection:
+    traffic_light: TrafficLight
```

Comment:
There are supposed to be 4 traffic lights per intersection? So should that be a list of `TrafficLight`, rather than one?